tests/printer/printer.cpp

```diff
@@ -768,3 +768,138 @@ TEST(Printer, noChangeToManualIds)
+    component2->appendMath(mathFooter);
+
+    auto component3 = libcellml::Component::create("component3");
+    model->addComponent(component3);
+    std::string equation3 =
+        "<apply><eq/>"
+        "<apply><diff/>"
+        "<bvar><ci>t</ci></bvar>"
+        "<ci>X</ci>"
+        "</apply>"
+        "<apply><minus/>"
+        "<apply><times/>"
+        "<ci>alpha_X</ci>"
+        "<apply><minus/>"
+        "<cn cellml:units=\"dimensionless\">1</cn>"
+        "<ci>X</ci>"
+        "</apply>"
+        "</apply>"
+        "<apply><times/>"
+        "<ci>beta_X</ci>"
+        "<ci>X</ci>"
+        "</apply>"
+        "</apply>"
+        "</apply>";
+    component3->setMath(mathHeader);
+    component3->appendMath(equation3);
+    component3->appendMath(mathFooter);
+
+    auto component4 = libcellml::Component::create("component4");
+    model->addComponent(component4);
+    std::string equation4 =
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\"><apply><eq/><apply><diff/><bvar><ci>t</ci></bvar><ci>X</ci></apply><apply><minus/><apply><times/><ci>alpha_X</ci><apply><minus/><cn cellml:units=\"dimensionless\">1</cn><ci>X</ci></apply></apply><apply><times/><ci>beta_X</ci><ci>X</ci></apply></apply></apply></math>";
+    component4->setMath(equation4);
+
+    auto component5 = libcellml::Component::create("component5");
+    model->addComponent(component5);
+    std::string equation5 =
+        "<math xmlns=\"http://www.w3.org/1998/Math/MathML\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\"><apply>          "
+        "<eq/><apply><diff/><bvar><ci>t</ci></bvar>\n\n<ci>X</ci></apply>"
+        "<apply><minus/><apply><times/><ci>     alpha_X</ci>             \n<apply><minus/>\n\n\n"
+        "<cn cellml:units=\"dimensionless\">1</cn><ci>X</ci></apply></apply><apply><times/>"
+        "<ci>beta_X</ci><ci>X</ci></apply>\n\n\n</apply></apply></math>";
+    component5->setMath(equation5);
+
+    auto component6 = libcellml::Component::create("component6");
+    model->addComponent(component6);
+    std::string equation6 =
+        "<apply><eq/>\n"
+        "  <apply><diff/>\n"
+        "    <bvar><ci>t</ci></bvar>\n"
+        "    <ci>X</ci>\n"
+        "  </apply>\n"
+        "  <apply><minus/>\n"
+        "    <apply><times/>\n"
+        "      <ci>alpha_X</ci>\n"
+        "      <apply><minus/>\n"
+        "        <cn cellml:units=\"dimensionless\">1</cn>\n"
+        "        <ci>X</ci>\n"
+        "      </apply>\n"
+        "    </apply>\n"
+        "    <apply><times/>\n"
+        "      <ci>beta_X</ci>\n"
+        "      <ci>X</ci>\n"
+        "    </apply>\n"
+        "  </apply>\n"
+        "</apply>\n";
+    component6->setMath(mathHeader);
+    component6->appendMath(equation6);
+    component6->appendMath(mathFooter);
+
+    std::string expectedOutput = fileContents("printer/pretty_print_mathml.cellml");
+    EXPECT_EQ(expectedOutput, printer->printModel(model));
+}
```

Comment:
I would probably split this test in six (one for each type of pretty printing you want to test) and maybe have the expected result in the `tests/printer/printer.cpp` rather than in `tests/printer/pretty_print_mathml.cellml`? Will leave it to @hsorby to comment on this.

Reply:
Yeah, I thought about that. But it's a big file and I couldn't be bothered doing all the line endings!  I guess if it's split into several tests they all make the same thing anyway.

Reply:
If you were to break things down, you would need to specify the expected output only once (and use it in all six tests) and it would be smaller, easier to understand, maintain (if ever needed), etc.

Reply:
Done in b43279a